kimchi/src/circuits/expr.rs

```diff
@@ -288,11 +289,12 @@ pub enum ConstantTerm<F> {
     Literal(F),
 }
 
-pub trait Literal: Sized {
+pub trait Literal: Sized + Clone {
```

Comment:
This is such a weird trait. Very unclear what it's doing without docs. The type suggests it's just some kind of combination of `To + TryTo + From`, but `Result<Self::F, Self>` is definitely not fitting the pattern then :thinking: .